packages/studio/src/components/ComponentTree.tsx

```diff
@@ -167,32 +179,119 @@ function useTree(): NodeModel<ComponentState>[] | undefined {
 }
 
 function useDropHandler() {
-  const updateComponentTree = useStudioStore(
-    (store) => store.actions.updateComponentTree
-  );
+  const [selectedComponentUUIDs, componentTree, updateComponentTree] =
+    useStudioStore((store) => {
+      return [
+        store.pages.selectedComponentUUIDs,
+        store.actions.getComponentTree(),
+        store.actions.updateComponentTree,
+      ];
+    });
 
-  const handleDrop = useCallback(
-    (tree: NodeModel<ComponentState>[]) => {
-      const updatedComponentTree = tree.map((n) => {
-        if (!n.data) {
-          throw new Error(
-            "Unable to handle drag and drop event in ComponentTree: " +
-              "No data passed into NodeModel<ComponentState> for node " +
-              JSON.stringify(n, null, 2)
-          );
-        }
-        const componentState: ComponentState = {
-          ...n.data,
-          parentUUID: n.parent.toString(),
-        };
-        if (componentState.parentUUID === ROOT_ID) {
-          delete componentState.parentUUID;
+  const handleSelectedDrop = useCallback(
+    (
+      updatedComponentTree: ComponentState[],
+      dragSourceId: string,
+      destinationParentId: string
+    ) => {
+      if (!componentTree) {
+        throw new Error(
+          "Unable to handle drag and drop event: component tree is undefined."
+        );
+      }
+      if (
+        isEqual(
```

Comment:
could you bring this out into a variable so it look prettier?  : P

Reply:
updated!